### PR TITLE
Weather app routing

### PR DESCRIPTION
**Pull request by dani-marcu**
test

### CODE REVIEW

File: angular/weather-fe/src/app/weather/weather.component.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  messageOne = '';
+  messageTwo = '';
+
+  onNewForecast(){
+    const loc = this.location
+    this.location = ''
+    fetch('http://localhost:3000/weather?address=' + loc).then((response) => {
+      response.json().then((data) => {
+        if (data.error) {
+          this.messageOne = data.error;
```
Comment (stevefai) [inline angular/weather-fe/src/app/weather/weather.component.ts:26]: don't use the same variable to represent different things. It's hard to know what the variable represents and it's also difficult to track how the variable changes. Additionally, the implementation is not very future proof: imagine at some point, you want the error message to be styled differently than the "success message" containing the address. At that point, it's much easier to treat the two messages separately if we already have two different variables.

in short: add an `error` variable.

File: angular/weather-fe/src/app/weather/weather.component.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+          this.messageTwo = '';
```
Comment (stevefai) [inline angular/weather-fe/src/app/weather/weather.component.ts:27]: I think it would be cleaner and more user friendly to always clear the `address` and `forecast` messages BEFORE fetching the data. This way, we also have the potential to add a "loading spinner" or an intermediate "fetching..." message in between calling the server and receiving the response.

**Review by stevefai — CHANGES_REQUESTED** [review]

**Review by stevefai — APPROVED** [review]
